### PR TITLE
 *: remove history_idx, squash rules for ref_commits 

### PR DESCRIPTION
**Pull request by erizocosmico**
Depends on #280

### CODE REVIEW

**Comment by erizocosmico** [comment]
Rebased and fixed conflicts

File: squash_iterator.go [hunk 1]
```diff
@@ -351,8 +352,8 @@ type squashRefIter struct {
 
 // NewAllRefsIter returns an iterator that will return all references
 // that match the given filters.
-func NewAllRefsIter(filters sql.Expression) RefsIter {
-	return &squashRefIter{filters: filters}
+func NewAllRefsIter(filters sql.Expression, virtual bool) RefsIter {
```
Comment (ajnavarro) [inline squash_iterator.go]: Can you add what `virtual` means to the doc?

Comment (erizocosmico) [reply squash_iterator.go]: Yes

**Comment by erizocosmico** [comment]
Fixed @ajnavarro

**Pull request merged**